### ISSUE TITLE
Enhance WSLENV to ignore translation failures when running Win32 commands from WSL

### ISSUE THREAD

**Issue by andrewluotechnologies** [Microsoft/WSL#3725]
If I run the command `WSLENV='PATH/l' cmd.exe /c echo hello` from WSL, I get a bunch of errors printed out on stderr:

> <3>init: (13347) ERROR: UtilTranslatePathList:2021: Failed to translate /usr/local/sbin
> <3>init: (13347) ERROR: UtilTranslatePathList:2021: Failed to translate /usr/local/bin
> <3>init: (13347) ERROR: UtilTranslatePathList:2021: Failed to translate /usr/sbin
> <3>init: (13347) ERROR: UtilTranslatePathList:2021: Failed to translate /usr/bin
> <3>init: (13347) ERROR: UtilTranslatePathList:2021: Failed to translate /sbin
> <3>init: (13347) ERROR: UtilTranslatePathList:2021: Failed to translate /bin
> <3>init: (13347) ERROR: UtilTranslatePathList:2021: Failed to translate /usr/games
> <3>init: (13347) ERROR: UtilTranslatePathList:2021: Failed to translate /usr/local/games
> <3>init: (13347) ERROR: UtilTranslatePathList:2021: Failed to translate /snap/bin

It would be nice to be able to disable these.  For example, if we could set an environment variable, such as 
`WSLENVIGNORETRANSLATIONERRORS=PATH:VAR2` that would suppress these messages.

Redirecting stderr is not an option because the Win32 program we call may be printing meaningful information on stderr.

For a single program this is a minor annoyance, but if you are calling a lot of programs in a script (for example a Makefile) you will get this for every Win32 program you call, resulting in a huge amount of output.  It also interferes with programmatically parsing stderr in bash scripts (of course we can workaround this but it's not ideal)

**Comment by 1ftskank** [comment]
Yes, please implement this as it makes using any windows executables from WSL unbearable!

**Comment by benhillis** [comment]
Good suggestion, I'll take care of this.

**Comment by JK-97** [comment]
at first ,my issue is when i run `wsl`, it will print the message  "An error occurred mounting one of your file systems. Please run 'dmesg' for more details.",
and run `dmesg`  in ubuntu ,it appear 

```
C:\Users\Benjamin>wsl
安装其中一个文件系统时出现错误。有关详细信息，请运行'dmesg'。
```
```
benjamin@DESKTOP-VGHN1LK:/mnt/c/Users/Benjamin$ dmesg
........
[  255.694326] init: (21) ERROR: UtilTranslatePathList:2721: Failed to translate C:\WINDOWS\system32
[  255.694383] init: (21) ERROR: UtilTranslatePathList:2721: Failed to translate C:\WINDOWS
[  255.694437] init: (21) ERROR: UtilTranslatePathList:2721: Failed to translate C:\WINDOWS\System32\Wbem
[  255.694491] init: (21) ERROR: UtilTranslatePathList:2721: Failed to translate C:\WINDOWS\System32\WindowsPowerShell\v1.0\

```




i just solved this problem， it will make the message   disappear， i had  instaled docker desktop wsl 2, so i run `wsl -l `

```
C:\Users\Benjamin>wsl -l
docker-desktop-data (defalut)
Ubuntu-18.04
docker-desktop
```

we can see default is  docker-desktop-data , so i use Ubuntu-18.04 replace it ,  `wsl --set-default Ubuntu-18.04`,  lastly , that message do not appear when i run `wsl`

when i run `wsl` ,it will attach in ubuntu, and run `dmesg`,we can see the output :

```
benjamin@DESKTOP-VGHN1LK:/mnt/c/Users/Benjamin$ dmesg
.........
[   14.412765] init: (2) ERROR: UtilCreateProcessAndWait:635: /bin/mount failed with 2
[   14.412863] init: (1) ERROR: UtilCreateProcessAndWait:655: /bin/mount failed with status 0x
[   14.412864] ff00
[   14.412866] init: (1) ERROR: ConfigMountFsTab:2162: Processing fstab with mount -a failed.
[   14.413056] init: (3) ERROR: UtilCreateProcessAndWait:635: /bin/mount failed with 2
[   14.413119] init: (1) ERROR: UtilCreateProcessAndWait:655: /bin/mount failed with status 0
```
obviously ，ERROR: UtilTranslatePathList:2661 disappear. but UtilCreateProcessAndWait:635 appear.

**Comment by PavelSosin-320** [comment]
This issue prevents my Ubuntu WSL2 from starting. I would like to use some stuff available cross-distro to avoid the building of self-containing packages for every type of distro. I'm looking at the calendar at my screen's right bottom corner and see 20/02/2021.  May I hope for a solution?

**Comment by surabibio** [comment]
It is Sept 2021 and still problem persists

**Comment by Satalicious** [comment]
i'm getting this error when i launch Ubuntu:

> <3>init: (203) ERROR: UtilTranslatePathList:2732: Failed to translate <somepath>

`ver`
> Microsoft Windows [Version 10.0.22621.160]

**Comment by pyhedgehog** [comment]
Right now problem seems gone and `/usr/bin` translates to `\\wsl.localhost\DISTRONAME\usr\bin`. Need to check what will happens in execution of `wsl -d d1 cmd.exe /c wsl -d d2`.
